src/client-library/deriv-api-client.ts

```diff
@@ -274,12 +274,12 @@ export class DerivAPIClient {
         await this.reinitializeData(this.subscribeHandler, this.authorizePayload);
     }
 
-    isSocketClosingOrClosed() {
-        return ![2, 3].includes(this.websocket.readyState);
+    isSocketReadyForSending() {
```

Comment:
```suggestion
    isSocketReady() {
```
just a nit - the rest lgtm